src/lib/utils/getReversedHighlight.ts

```diff
@@ -0,0 +1,33 @@
+import unescape from './unescape';
+
+const getReversedHighlight = (attribute: string) => {
```

Comment:
I think that the Autocomplete way of doing it (inspired by the way React InstantSearch does it) is more API-based, and makes more sense rather than using Regex.

1. We first parse the attribute to have the highlighted parts. This step would be valuable in InstantSearch.js to then expose the highlighted parts to Vue InstantSearch and Angular InstantSearch (so that do don't rely on unsafe XSS APIs): https://github.com/algolia/autocomplete.js/blob/547719b1e59f76474ad68e2b65dbefb3d64772be/packages/autocomplete-preset-algolia/src/highlight/parseAttribute.ts#L18
2. We reverse these matches to get the reverse highlighted matches _with the same API_: https://github.com/algolia/autocomplete.js/blob/next/packages/autocomplete-preset-algolia/src/highlight/reverseHighlightedParts.ts#L3
3. We expose the API to our renderers: https://github.com/algolia/autocomplete.js/blob/next/packages/autocomplete-preset-algolia/src/highlight/parseAlgoliaHitReverseHighlight.ts#L11
4. Our renderers use this API to speak the right language given the framework (here, vanilla JavaScript): https://github.com/algolia/autocomplete.js/blob/547719b1e59f76474ad68e2b65dbefb3d64772be/packages/autocomplete-js/src/highlight.ts#L39-L54

Reply:
it would indeed be interesting to get an array of parts (regardless of which attribute it comes from), and using the highlight pre and post tag to split it. Then you can map over them to turn them into highlighted or non-highlighted parts (then joining up again for plain html)

Reply:
It was actually the first method I was doing but I thought it was a bit complex/overkill compared to simply reversing the highlighted parts.

I'll give it an other try :D

Reply:
I think I didn't get what it was at first place because now that it's almost done, I feel dumb thinking that it was overkill, doing tests and I'll push it ASAP!

Reply:
@francoischalifour in this method, you don't escape the highlight on non alphanumeric characters, right? 

Should I implement it to match `highlight` behaviour?

Reply:
We escape these characters by default: https://github.com/algolia/autocomplete.js/blob/547719b1e59f76474ad68e2b65dbefb3d64772be/packages/autocomplete-preset-algolia/src/highlight/parseAttribute.ts#L5-L11

The way we do it in Autocomplete might differ a bit from InstantSearch, but we shouldn't change how it's done in IS.js now since it'll probably increase the scope of this feature.